pkg/handlers/scan_test.go

```diff
@@ -458,3 +463,484 @@ func TestScanHandlers_GetScanOperatingSystemVulnerabilitiesByID(t *testing.T) {
+					return nil, customerrors.ErrScanNotFound
+				},
+			},
+			vulnService:         &mock_services.MockVulnerabilityService{},
+			scanScheduleService: &mock_services.MockScanScheduleService{},
+			hostService:         &mock_services.MockHostService{},
+			emailService:        &mock_services.MockEmailService{},
+			eventBus:            &events.NatsEventBus{},
+			r: func() *http.Request {
+				r := httptest.NewRequest("GET", "/api/scans/"+scanID.String()+"/services/"+strconv.Itoa(serviceID)+"/vulnerabilities", nil)
+				r.SetPathValue("id", scanID.String())
+				r.SetPathValue("service_id", strconv.Itoa(serviceID))
+				r = r.WithContext(context.WithValue(r.Context(), middleware.ContextRoles, []domain.Role{domain.RoleAdmin}))
+				return r
+			}(),
+			wantStatus:       http.StatusNotFound,
+			wantBodyContains: []string{},
+		},
+		{
+			name: "Scan Not Completed → 409",
+			scanService: &mock_services.MockScanService{
+				MockGetScanByID: func(ctx context.Context, id uuid.UUID) (*domain.Scan, error) {
+					return &domain.Scan{Status: "InProgress"}, nil
+				},
+			},
+			vulnService:         &mock_services.MockVulnerabilityService{},
+			scanScheduleService: &mock_services.MockScanScheduleService{},
+			hostService:         &mock_services.MockHostService{},
+			emailService:        &mock_services.MockEmailService{},
+			eventBus:            &events.NatsEventBus{},
+			r: func() *http.Request {
+				r := httptest.NewRequest("GET", "/api/scans/"+scanID.String()+"/services/"+strconv.Itoa(serviceID)+"/vulnerabilities", nil)
+				r.SetPathValue("id", scanID.String())
+				r.SetPathValue("service_id", strconv.Itoa(serviceID))
+				r = r.WithContext(context.WithValue(r.Context(), middleware.ContextRoles, []domain.Role{domain.RoleAdmin}))
+				return r
+			}(),
+			wantStatus:       http.StatusConflict,
+			wantBodyContains: []string{},
+		},
+		{
+			name: "Bad ServiceID → 400",
+			scanService: &mock_services.MockScanService{
+				MockGetScanByID: func(ctx context.Context, id uuid.UUID) (*domain.Scan, error) {
+					return &domain.Scan{Status: "Completed"}, nil
+				},
+			},
+			vulnService:         &mock_services.MockVulnerabilityService{},
+			scanScheduleService: &mock_services.MockScanScheduleService{},
+			hostService:         &mock_services.MockHostService{},
+			emailService:        &mock_services.MockEmailService{},
+			eventBus:            &events.NatsEventBus{},
+			r: func() *http.Request {
+				r := httptest.NewRequest("GET", "/api/scans/"+scanID.String()+"/services/invalid/vulnerabilities", nil)
+				r.SetPathValue("id", scanID.String())
+				r.SetPathValue("service_id", "invalid")
+				r = r.WithContext(context.WithValue(r.Context(), middleware.ContextRoles, []domain.Role{domain.RoleAdmin}))
+				return r
+			}(),
+			wantStatus:       http.StatusBadRequest,
+			wantBodyContains: []string{},
+		},
+		{
+			name: "ScanID exists but Not ServiceID for that Scan → 404",
```

Comment:
`[Nit]` Spelling can be improved:

```suggestion
			name: "Service not found for scan → 404",
```